odoo/addons/base/models/ir_model.py

```diff
@@ -1408,10 +1409,16 @@ def _reflect_inherits(self, model_names):
                 ] + [
                     (model_id, get_model_id(parent_name), get_field_id(field))
                     for parent_name, field in cls._inherits.items()
+                ] + [
+                    (model_id, get_model_id(field.comodel_name), get_field_id(field_name))
+                    for (field_name, field) in inspect.getmembers(cls)
+                    if isinstance(field, fields.Field)
+                    if field.type == 'many2one' and not field.related and field.delegate
+                    if field_name not in cls._inherits.values()
```

Comment:
LGTM for stable

But I feel this logic should be done in in `MetaModel.__init__` to complete missing `_inherits`. So that we even don't need the code in  [`_inherits_check`](https://github.com/odoo-dev/odoo/blob/945c09cfcf8774424944d8e04fd3b92be11e85cd/odoo/models.py#L3457)

Or we can make `field.delegate` not configurable for developers and force them to call `_inherits` explicitly.